enhancements/cloud-controller-manager/openstack-cloud-controller-manager.md

```diff
@@ -0,0 +1,218 @@
+
+**Note:** Example of manual testing: https://asciinema.org/a/303399?speed=2
+
+#### Write Cluster Cloud Controller Manager Operator
+
+The operator should be able to create, configure and manage different CCMs (including one for OpenStack) in OpenShift.
+
+Actions:
+
+- Create a new repo in OpenShift’s github: https://github.com/openshift/cluster-cloud-controller-manager-operator
+
+- Implement the operator, using [library-go](https://github.com/openshift/library-go) primitives.
+
+#### Build the operator image by OCP automation
+
+To start using CCM operator in OpenShift we need to build its image and make sure it is a part of the OpenShift release image. The image should be automatically tested before it becomes available. Dockerfile should be a part of the operator's repo.
+
+Actions:
+
+- Configure CI operator to build CCM operator image.
+
+CI operator will run containerized and End-to-End tests and also push the resulting image in the OCP Quay account.
+
+#### Integrate the solution with OpenShift
+
+Actions:
+
+- Make sure Cinder CSI driver is supported in OpenShift.
+
+- Add CCM operator support in Cluster Version Operator. CCM operator will be deployed on early stages of installation and then it deploys and configures OpenStack CCM itself.
+
+- Change Kubelet configuration for OpenStack in [Machine Config Operator templates](https://github.com/openshift/machine-config-operator/blob/d044c74ea4b9900c269ee8de8131ed49ba6fedc8/templates/master/01-master-kubelet/openstack/units/kubelet.service.yaml#L30) to adopt external cloud provider.
+
+- Change Config observer in [library-go](https://github.com/openshift/library-go/blob/16d6830d0b80dc2a3315207116d009ed2dd4cebf/pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go#L154) in the same manner. Then the library should be bumped in `cluster-kube-apiserver-operator` and `cluster-kube-controller-manager-operator`.
+
+### Cloud Controller Manager installation workflow
+
+Starting from OpenShift 4.7 the installation of OpenShift on OpenStack will be next:
+
+- All Kubernetes components that require cloud provider functionality (`kubelet`, `kube-apiserver`, `kube-controller-manager`) are configured to use `external` cloud provider. In other words, they are launched with the "--cloud-provider external" option.
+
+- `cluster-version-operator` starts `cluster-cloud-controller-manager-operator`.
+
+- `cluster-cloud-controller-manager-operator` checks if it runs on OpenStack, populates [configuration](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/using-openstack-cloud-controller-manager.md#config-openstack-cloud-controller-manager) and secrets for OpenStack Cloud Controller Manager, creates [OpenStack CCM deployment](https://github.com/kubernetes/cloud-provider-openstack/blob/master/manifests/controller-manager/openstack-cloud-controller-manager-ds.yaml) and monitors its status.
+
+### Risks and Mitigations
```

Comment:
+1 for identifying risks.  Minimally for the Med/Likely risks we should come up with mitigation plans.